src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,349 @@
+  supportedVMArchOptions,
+  CANCEL_UPLOAD_REASON,
+} from "util/uploadExternalFormatFile";
+import { getInstanceName } from "util/operations";
+import classnames from "classnames";
+import InstanceFileTypeSelector, {
+  InstanceFileType,
+} from "./InstanceFileTypeSelector";
+
+interface Props {
+  close: () => void;
+  uploadState: UploadState | null;
+  setUploadState: (state: UploadState | null) => void;
+  fileType: InstanceFileType;
+  setFileType: (value: InstanceFileType) => void;
+  defaultInstanceName?: string;
+}
+
+const UploadExternalFormatFileForm: FC<Props> = ({
+  close,
+  uploadState,
+  setUploadState,
+  fileType,
+  setFileType,
+  defaultInstanceName,
+}) => {
+  const { project, isLoading } = useProject();
+  const instanceNameAbort = useState<AbortController | null>(null);
+  const [socket, setSocket] = useState<WebSocket>();
+  const [operationId, setOperationId] = useState("");
+  const toastNotify = useToastNotification();
+  const notify = useNotify();
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+  const { data: settings } = useSettings();
+
+  const handleUploadProgress = (
+    instanceName: string,
+    current: number,
+    total: number,
+  ) => {
+    setUploadState({
+      percentage: Math.floor((current / total) * 100),
+      loaded: current,
+      total: total,
+    });
+
+    if (current === total) {
+      close();
+      toastNotify.info(
+        <>
+          Upload completed. Now creating instance{" "}
+          <strong>{instanceName}</strong>.
+        </>,
+      );
+      navigate(`/ui/project/${project?.name}/instances`);
+    }
+  };
+
+  const handleUploadError = (error: Error) => {
+    notify.failure("Image upload failed.", error);
+    setUploadState(null);
+  };
+
+  const handleCancelUpload = () => {
+    if (socket && socket.readyState === WebSocket.OPEN) {
+      socket.close(1000, CANCEL_UPLOAD_REASON);
+    }
+    eventQueue.remove(operationId);
+  };
+
+  const handleSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created instance{" "}
+        <InstanceLink
+          instance={{ project: project?.name || "", name: instanceName }}
+        />
+        .
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const invalidateCache = () => {
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+  };
+
+  const handleSubmit = (values: UploadExternalFormatFileFormValues) => {
+    if (!values.imageFile) {
+      return;
```

Comment:
this is a unlikely condition since form submission is disabled if there is no file selected. I think I previously added this here to get around type checking, probably with some logic before the api call. I see that is no longer needed though, so I'd rather remove this condition than adding a notification.